### PR TITLE
Random cleanup 2

### PR DESCRIPTION
**Pull request by kamilritz**
This PR contains 8 different clean-ups each described in the commit message.

I recommend to review this commit by commit.

And it saves 450 Bytes of flash.

SITL Mission test with strange behavior: [Here](https://review.px4.io/plot_app?log=c8dc5c5d-b29b-48a8-b074-9a04aa7d8097) (SOLVED)

https://github.com/PX4/Matrix/pull/117 is solving the issue. New SITL Mission test [log](https://review.px4.io/plot_app?log=92f091db-0a76-4193-ae74-70e924a78bdf)

### CODE REVIEW

File: EKF/estimator_interface.cpp [hunk 1]
```diff
@@ -367,22 +367,20 @@ void EstimatorInterface::setOpticalFlowData(uint64_t time_usec, flow_message *fl
 		// check if enough integration time and fail if integration time is less than 50%
 		// of min arrival interval because too much data is being lost
 		float delta_time = 1e-6f * (float)flow->dt; // in seconds
-		float delta_time_min = 5e-7f * (float)_min_obs_interval_us;
+		float delta_time_min = 0.5f * 1e-6f * (float)_min_obs_interval_us;
```
Comment (bresch) [inline EKF/estimator_interface.cpp]: `0.5e-6f` ?
could also be const floats

File: EKF/control.cpp [hunk 2]
```diff
@@ -801,36 +801,29 @@ void Ekf::controlHeightSensorTimeouts()
...
 
-				// reset the height mode
 				setControlGPSHeight();
 
 				// request a reset
 				reset_height = true;
 				ECL_WARN_TIMESTAMPED("baro hgt timeout - reset to GPS");
 
-			} else if (reset_to_baro) {
+			} else if (baro_data_available) {
```
Comment (bresch) [inline EKF/control.cpp]: you can also remove all the 4
```c++
} else {
   // we have nothing we can reset to
   // deny a reset
   reset_height = false;
}
```
as the reset_height variable is already initialized to `false`

File: EKF/estimator_interface.cpp (hunk 1 above)
Comment (kamilritz) [reply EKF/estimator_interface.cpp]: only delta_time_min can be const

**Comment by kamilritz** [comment]
@bresch I ran a SITL square mission test. The line tracking and estimation is not that great, is this normal?

File: EKF/control.cpp (hunk 2 above)
Comment (kamilritz) [reply EKF/control.cpp]: Nice idea

File: EKF/vel_pos_fusion.cpp [hunk 3]
```diff
@@ -93,29 +93,29 @@ bool Ekf::fuseVerticalVelocity(const Vector3f &innov, const Vector2f &innov_gate
...
-			if (!_fuse_hpos_as_odom) {
-				_time_last_hor_pos_fuse = _time_last_imu;
-
-			} else {
-				_time_last_delpos_fuse = _time_last_imu;
-			}
-			_innov_check_fail_status.flags.reject_hor_pos = false;
+	innov_var(0) = P[7][7] + obs_var(0);
+	innov_var(1) = P[8][8] + obs_var(1);
+	test_ratio(0) = fmaxf( sq(innov(0)) / (sq(innov_gate(0)) * innov_var(0)),
```
[the remark this answers is not in the document]
Comment (kamilritz) [reply EKF/vel_pos_fusion.cpp]: Done

**Comment by kamilritz** [comment]
Commit "Improve matrix library usage" introduced some error. Investigation ongoing.

File: EKF/control.cpp [hunk 4]
```diff
@@ -796,106 +789,79 @@ void Ekf::controlHeightSensorTimeouts()
...
 			// if baro data is acceptable and GPS data is inaccurate, reset height to baro
-			bool reset_to_baro = baro_data_consistent && baro_data_fresh && !_baro_hgt_faulty && !gps_hgt_accurate;
-
-			// if GPS height is unavailable and baro data is available, reset height to baro
-			reset_to_baro = reset_to_baro || (_gps_hgt_intermittent && baro_data_fresh);
-
-			// if we cannot switch to baro and GPS data is available, reset height to GPS
-			bool reset_to_gps = !reset_to_baro && !_gps_hgt_intermittent;
+			const bool reset_to_baro = baro_data_fresh &&
+						   ( (baro_data_consistent && !_baro_hgt_faulty && !gps_hgt_accurate) ||
```
Comment (bresch) [inline EKF/control.cpp]:
```suggestion
						   ((baro_data_consistent && !_baro_hgt_faulty && !gps_hgt_accurate) ||
```

File: EKF/control.cpp [hunk 5]
```diff
@@ -796,106 +789,79 @@ void Ekf::controlHeightSensorTimeouts()
...
+						     _gps_hgt_intermittent );
```
Comment (bresch) [inline EKF/control.cpp]:
```suggestion
						     _gps_hgt_intermittent);
```

File: EKF/ekf_helper.cpp [hunk 6]
```diff
@@ -1664,9 +1657,7 @@ void Ekf::calcExtVisRotMat()
 
 	}
 
-	// convert filtered vector to a quaternion and then to a rotation matrix
-	q_error.from_axis_angle(_ev_rot_vec_filt);
-	_ev_rot_mat = Dcmf(q_error); // rotation from EV reference to EKF reference
+	_ev_rot_mat = Dcmf(_ev_rot_vec_filt); // rotation from EV reference to EKF reference
```
[the remark this answers is not in the document]
Comment (kamilritz) [reply EKF/ekf_helper.cpp]: i changed it to `_R_ev_to_ekf`

**Comment by kamilritz** [comment]
updated

**Comment by kamilritz** [comment]
Flight Tests passed, they can be found here: https://github.com/PX4/Firmware/pull/13870

**Pull request merged**